### PR TITLE
[Fix] Heat Rising Infinitely Fix

### LINKED ISSUE #1541: 4fps "Yep, something is wrong with your temperature: Infinity/NaN."
At some point during run-time game throws a trillion of Debug.Logs "Yep, something is wrong with your temperature: **Infinity**." Alternating with some amount of  "Yep, something is wrong with your temperature: **NaN**." And then only the **NaN** ones. FPS drops as expected.

Not sure if it's a bug, or just not fully implemented temperature system.

untitled
...I placed heater after it happened.

### PR DESCRIPTION
**Pull request by Geoffrotism**
Fixes #1541 
Closes #1541 
Previously the temperature scaled infinitely. This fixes the code that caused this and added a check to prevent it from accidentally occurring in the future.

Also balances the flow of heat throughout the ship.

### CODE REVIEW

File: Assets/Scripts/Models/Area/Temperature.cs [hunk 1]
```diff
@@ -339,14 +348,36 @@ private void ForwardTemp()
...
+                    if (z < sizeZ - 1)
                     {
                         temp_curr[index] += c * 0.5f * (temp_old[index_above] - temp_old[index]);
+                    }
+
+                    // Add a little bit more flow to the temperature.
+                    float value = temp_curr[index];
+
+                    // FINE tune the below number. ".005" has a huge affect.
+                    value += 0.065f * value;
```
Comment (koosemose) [inline Assets/Scripts/Models/Area/Temperature.cs]: effect ;)

File: Assets/Scripts/Models/Area/Temperature.cs [hunk 2]
```diff
@@ -292,7 +292,7 @@ private void ForwardTemp()
             for (int y = 0; y < sizeY; y++) 
             {
                 for (int x = 0; x < sizeX; x++) 
-                {
+                {
```
Comment (koosemose) [inline Assets/Scripts/Models/Area/Temperature.cs]: You seem to have picked up some extra spaces here

File: Assets/Scripts/Models/Area/Temperature.cs [hunk 3]
```diff
@@ -258,17 +258,17 @@ private int GetIndex(int x, int y, int z)
...
         {
             foreach (Action<float> act in sinksAndSources.Values) 
             {
                 act(deltaT);
             }
-        }
-
-        Thread thread = new Thread(ForwardTemp);
-        thread.Start();
+        }
```
Comment (koosemose) [inline Assets/Scripts/Models/Area/Temperature.cs]: Extra Spaces here as well

File: Assets/Scripts/Models/Area/Temperature.cs [hunk 4]
```diff
@@ -301,14 +301,23 @@ private void ForwardTemp()
...
+                    if ((tile.Room == null || tile.Room.ID == 0) &&
+                        tile.Furniture == null) 
                     { 
+                        temp_curr[index] = 0f;
                         temp_old[index] = 0f;
+                        continue;
+                    }
+
+                    // If this tile has an object that gives off heat AND is hotter than the surrounding tiles, skip it.
+                    // TODO: Add the second part of the above statement.
```
Comment (koosemose) [inline Assets/Scripts/Models/Area/Temperature.cs]: Lots of trailing spaces here as well

File: Assets/Scripts/Models/Area/Temperature.cs [hunk 5]
```diff
@@ -301,14 +301,23 @@ private void ForwardTemp()
...
                     int index_below = GetIndex(x, y, z - 1);
                         
-                    // Update temperature using finite difference and forward method:
-                    // U^{n+1} = U^n + dt*(\Div alpha \Grad U^n).
                     temp_curr[index] = temp_old[index];
 
-                    // If empty space, set temperature to 0.                 
-                    if (WorldController.Instance.GetTileAtWorldCoord(new Vector3(x, y, z)).Room == null) 
+                    // If empty space, set temperature to 0.     
+                    Tile tile = WorldController.Instance.GetTileAtWorldCoord(new Vector3(x, y, z));
```
Comment (koosemose) [inline Assets/Scripts/Models/Area/Temperature.cs]: More excess spaces

File: Assets/StreamingAssets/Data/Furniture.xml [hunk 6]
```diff
@@ -655,6 +656,8 @@
         <ContextMenuAction FunctionName="Vent_Open" Text="Open Vent" RequireCharacterSelected="false" />
         <ContextMenuAction FunctionName="Vent_Close" Text="Close Vent" RequireCharacterSelected="false" />
 
+
+
```
Comment (koosemose) [inline Assets/StreamingAssets/Data/Furniture.xml]: Lots of extra linebreaks here

File: Assets/Scripts/Models/Area/Temperature.cs (hunk 3 above)
Comment (Geoffrotism) [inline Assets/Scripts/Models/Area/Temperature.cs]: how do i catch these in the future?

Comment (koosemose) [inline Assets/Scripts/Models/Area/Temperature.cs]: No idea honestly... I just stumbled on the first two. I can fix it real quick with a script though, if you don't mind.

Comment (Geoffrotism) [inline Assets/Scripts/Models/Area/Temperature.cs]: all yours. will try to remember to catch these in the future

**Comment by koosemose** [comment]
Looks good, stylecop clean, and I didn't break anything removing spaces and linebreaks. Merged.

**Pull request merged**